### DISCUSSION TITLE
Pass environment variables to applications executed via nwg start menu/drawer

### DISCUSSION THREAD

**Discussion by Noctivivans** [nwg-piotr/nwg-shell#305] [Q&A] [answered]
Hi,

How can I pass environment variables to applications executed using drawer or start menu? To give you an example, when I run keepassxc, variables like QT_QPA_PLATFORMTHEME are ignored, resulting falling back to default theme. I've tried to set variables both for my shell (fish, but also tried bash with the same result) and in more generic way (.profile), but they are still being ignored (/proc/ shows that they are simply missing).
Thanks in advance :)

**Comment by nwg-piotr** [comment]
Hi,

As well the drawer, as the menu, execute what they find in appropriate .desktop files. You could copy the keepassxc's .desktop file to `~/.local/share/applications/` and edit it to add the variable in the `Exec=` line.

**Reply by Noctivivans** [reply]
Thanks for quick reply :) is there any way to set them globally, or should I edit .desktop files for all apps I use?

**Reply by nwg-piotr** [reply, marked as answer]
Editing either /etc/profile or /etc/environment should do the job. On Hyprland you can set env vars in the config file as well.
